modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AbstractAccessControlProvider.java

```diff
@@ -63,17 +67,19 @@ protected void initialize(AccessControlSchema config) {
    * graph}.
    *
    * @param group is the {@link AccessControlGroup} to check.
-   * @param visitedGroups is where the {@link AccessControlGroup} are collected to detect cycles.
+   * @param nodeCycle the {@link NodeCycle} used to detect cycles.
    */
-  protected void checkForCyclicDependencies(AccessControlGroup group, Set<AccessControlGroup> visitedGroups) {
+  protected void checkForCyclicDependencies(AccessControlGroup group, NodeCycle<AccessControlGroup> nodeCycle) {
 
-    boolean added = visitedGroups.add(group);
-    if (!added) {
-      // mmm NodeCycleException would be very helpful here...
-      throw new IllegalStateException("Cyclic inheritance of access control groups detected for " + group);
-    }
     for (AccessControlGroup inheritedGroup : group.getInherits()) {
-      checkForCyclicDependencies(inheritedGroup, visitedGroups);
+      List<AccessControlGroup> inverseCycle = nodeCycle.getInverseCycle();
```

Comment:
@maybeec just to clarify: you propose a renaming of `visitedGroups` to (for example) `vistiedStack`?